pages/api/student/index.ts

```diff
@@ -0,0 +1,40 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { ResponseUtil } from "@utils/responseUtil";
+import { createStudent } from "@database/student";
+import { CreateStudentInput } from "@models/Student";
+import { validateCreateStudent } from "@utils/validation/student";
+
+/**
+ * @param req API request object
+ * @param res API response object
+ */
+export default async function handle(
+    req: NextApiRequest,
+    res: NextApiResponse,
+): Promise<void> {
+    if (req.method == "POST") {
+        const input = req.body as CreateStudentInput;
+        const validationErrors = await validateCreateStudent(input);
+        if (validationErrors.length !== 0) {
+            ResponseUtil.returnBadRequest(res, validationErrors.join(", "));
+        } else {
+            const student = await createStudent(input);
+            if (!student) {
+                ResponseUtil.returnBadRequest(
+                    res,
+                    `Student could not be created`,
+                );
+                return;
+            }
+            ResponseUtil.returnOK(res, student);
```

Comment:
Should I return info about parent/parentId?

Reply:
do we need to return a response here? I think upon POST the client doesn't really need any info back except that  it went through

Reply:
Don't think we require it. I'll remove it then.

Reply:
Since other POST requests return a response, I'll merge it in with what I have right now. We can make the change later.